### PR TITLE
MAYA-109316 Propagate state of dirty draw items

### PR DESCRIPTION
**Pull request by JGamache-autodesk**
when they are about to be redrawn after getting out of sync.

This fixes drawing multiple Repr in sequence after dirtying the RPrim.
The first draw will clean the RPrim and the Reprs that were selected
to be drawn. Reprs that were not drawn will stay stale and need to
signal the Sync engine that they need to be refreshed on the next
draw cycle that will draw them.

### CODE REVIEW

**Review by williamkrick — APPROVED** [review]
It looks good. More duplicated code means refactoring these and adding a common parent class more and more important. We've talked about that a lot in the past, and I just logged MAYA-109380 so we can hopefully put it on the schedule.

File: lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp [hunk 1]
```diff
@@ -648,18 +648,39 @@ HdDirtyBits HdVP2Mesh::_PropagateDirtyBits(HdDirtyBits bits) const
...
+            for (const HdRepr::DrawItemUniquePtr& item : items) {
+                if (HdVP2DrawItem* const drawItem = static_cast<HdVP2DrawItem*>(item.get())) {
 #endif
-                drawItem->SetDirtyBits(bits);
+                    drawItem->SetDirtyBits(bits);
+                }
+            }
+        }
+    } else {
+        // RPrim is clean, find out if any drawItem about to be shown is dirty:
```
Comment (kxl-adsk) [inline lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp:668]: What's the performance impact of this change?

**Comment by kxl-adsk** [comment]
FYI. This change will require fixes to compile with USD 20.02 (and 19.11 since we haven't yet cleaned it up)
```
E:\usd\maya-usd_pr\lib\mayaUsd\render\vp2RenderDelegate\mesh.cpp(746): error C2440: 'static_cast': cannot convert from 'const pxrInternal_v0_19__pxrReserved__::HdDrawItem *' to 'pxrInternal_v0_19__pxrReserved__::HdVP2DrawItem *'
```

File: lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp (hunk 1 above)
Comment (JGamache-autodesk) [reply lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp:672]: No performance impact. I checked and vertex buffers are correctly shared and are updated only once when changing a sphere radius. The geometries get updated for each displayed render item, but only as necessary to clean the dirty bits found in the reprs.

**Review by kxl-adsk — APPROVED** [review]

**Pull request merged**